addons/website/models/website_page.py

```diff
@@ -269,3 +285,150 @@ def action_page_debug_view(self):
+                    status=response.status,
+                    response=[response.response[0]],
+                )
+                self._post_process_response_from_cache(request, resp)
+                return resp
+            else:
+                response = self._get_response_raw(request)
+                self._get_response_cached.__cache__.add_value(self, request, cache_value=(response, cache_key))
+
+        return self._get_response_raw(request)
+
+    @tools.conditional(
+        'xml' not in tools.config['dev_mode'],
+        tools.ormcache('self._get_cache_key(request)', cache='templates.cached_values'),
+    )
+    def _get_response_cached(self, request) -> tuple[http.Response, int, str]:
+        cache_key = self._get_cache_key(request)
+        response = self._get_response_raw(request)
+        result = response, cache_key
+
+        if not response:
+            raise PageCanNotBeCached(result)
+
+        response.flatten()
+        if not self._allow_cache_insertion(response.response[-1]):
+            raise PageCanNotBeCached(result)
+
+        return result
+
+    def _get_response_raw(self, request) -> http.Response:
+        req_page = request.httprequest.path
+
+        # Force prefetch on all fetchable fields to get all data at once (e.g., 'website_meta').
+        fields_to_fetch = [name for name, field in self._fields.items() if field.store and field.column_type]
+        self.fetch(fields_to_fetch)
+
+        # Force prefetch on all fetchable fields to get all data at once (e.g., 'website_meta').
+        fields_to_fetch = [name for name, field in self.view_id._fields.items() if field.store and field.column_type]
+        self.view_id.fetch(fields_to_fetch)
+
+        if (
+            (self.env.user.has_group('website.group_website_designer') or self.is_visible)
+            and (
+                # If a generic page (niche case) has been COWed and that COWed
+                # page received a URL change, it should not let you access the
+                # generic page anymore, despite having a different URL.
+                self.website_id
+                or self.view_id.id == self.env['ir.ui.view'].with_context(website_id=request.website.id)._get_cached_template_info(self.view_id.key)['id']
+            )
+        ):
+            _, ext = os.path.splitext(req_page)
+            response = request.render(self.view_id.id, {
+                'main_object': self,
+            }, mimetype=EXTENSION_TO_WEB_MIMETYPES.get(ext, 'text/html'))
+            response.time = time.time()
+            return response
+        return False
+
+    @api.model
+    def _get_page_info(self, request) -> dict | None:
+        return self._get_page_info_cached(request)
+
+    @tools.conditional(
+        'xml' not in tools.config['dev_mode'],
+        tools.ormcache('request.httprequest.path', cache='templates.cached_values'),
+    )
+    @api.model
+    def _get_page_info_cached(self, request) -> dict | None:
```

Comment:
- I am confused, what's the point of having a function just calling another cached one like this?
- Did not we decide to not had stuff like `-> dict | None`?

Reply:
A refactoring/rebase... mistake.

If they are no page, they are None result